src/__tests__/api.test.ts

```diff
@@ -0,0 +1,29 @@
+import { buildDecoratorFactories, FACTORIES, PropertyDecoratorFactory } from '..';
+
+describe('buildDecoratorFactories', () => {
+    it('produces new decorator factories', () => {
+        const noop = () => { };
+        const transformer = () => noop;
```

Comment:
A transformer takes a decorator factory and returns a decorator factory. In this case, I'm totally ignoring the input always return a noop decorator.

Reply:
Would be nice to have a real (even if minimal) transformer here to make the test more realistic and also to provide a usage example